operator/src/main/java/oracle/kubernetes/operator/steps/WatchDomainIntrospectorJobReadyStep.java

```diff
@@ -33,15 +37,22 @@ public NextAction apply(Packet packet) {
 
     // No need to spawn a watcher if the job is already complete
     if (domainIntrospectorJob != null && !JobWatcher.isComplete(domainIntrospectorJob)) {
-      JobWatcher jw =
-          JobWatcher.create(
-              ThreadFactorySingleton.getInstance(),
-              namespace,
-              initialResourceVersion,
-              tuning,
-              new AtomicBoolean(false));
-
-      return doNext(jw.waitForReady(domainIntrospectorJob, getNext()), packet);
+      JobWatcher jw = null;
+      if (jws == null || null == jws.get(namespace)) {
+        jw =
+            JobWatcher.create(
+                ThreadFactorySingleton.getInstance(),
+                namespace,
+                initialResourceVersion,
+                tuning,
+                new AtomicBoolean(false));
```

Comment:
I see that you didn't write this line, but it was wrong before.  I purpose of passing in a boolean is so that the watcher can eventually be stopped if and when the operator stops watching this namespace.  The AtomicBoolean should be passed down just like the other use cases for Watchers.

Reply:
not sure i follow what you are saying here...

Reply:
is this better?